src/StochasticStyles/spawning.jl

```diff
@@ -10,97 +10,97 @@ Returns the value deposited.
     return projected_deposit!(valtype(w), w, add, val, parent, threshold)
 end
 # Non-integer
-@inline function projected_deposit!(::Type{T}, w, add, value, parent, thresh) where {T}
-    # ensure type stability.
-    threshold = T(thresh)
+@inline function projected_deposit!(
+    ::Type{T}, w, add, value, parent, threshold
+) where {T<:Real}
+    thresh = T(threshold)
     val = T(value)
-
     absval = abs(val)
-    if absval < threshold
-        if rand() < abs(val) / threshold
-            val = sign(val) * threshold
+    if absval < thresh
+        if rand() < absval / thresh
+            val = sign(val) * thresh
         else
-            val = zero(val)
+            val = zero(T)
         end
     end
     if !iszero(val)
         deposit!(w, add, val, parent)
     end
 
-    return abs(val)
+    return val
 end
 # Round to integer
-@inline function projected_deposit!(::Type{T}, w, add, val, parent, _) where {T<:Integer}
-    intval = T(sign(val)) * floor(T, abs(val) + rand())
-    if !iszero(intval)
-        deposit!(w, add, intval, parent)
+@inline function projected_deposit!(
+    ::Type{T}, w, add, val, parent, threshold=0
+) where {T<:Integer}
+    if !iszero(threshold)
+        throw(ArgumentError("Thresholding not supported for integer spawns"))
+    end
```

Comment:
Hmm... This get called through `spawn!(WithReplacement(), ...) (in `styles.jl` line 23, 54, 57). I'm pretty sure since `WithReplacement` is created right there and its threshold is zero, the check should get compiled away.

I'll do some benchmarks just in case to see if removing the check makes any difference.